util/ranger/range.go

```diff
@@ -626,3 +626,27 @@ func (r *Builder) BuildTableRanges(rangePoints []point) []types.IntColumnRange {
 	}
 	return tableRanges
 }
+
+func (r *Builder) buildColumnRanges(points []point, tp *types.FieldType) []*types.ColumnRange {
+	columnRanges := make([]*types.ColumnRange, 0, len(points)/2)
+	for i := 0; i < len(points); i += 2 {
+		startPoint := r.convertPoint(points[i], tp)
+		endPoint := r.convertPoint(points[i+1], tp)
```

Comment:
Is it always guaranteed that the `len(points)` is an even number? If not, `i+1` may equal to `len(points)`.